### ISSUE TITLE
Making `meta` template-parts more flexible

### ISSUE THREAD

**Issue by aurooba** [bacoords/block-theme#10]
One of the advantages of `template tags` is that you can plug them easily into custom templates, functions, or other programmatic ways of displaying posts. The way the `meta` template-parts are set up right now, that's not possible, because they _assume_ usage within The Loop. (For example: https://github.com/understrap/tangent/blob/develop/template-parts/meta-posted-on.php)

So, what if we adjust them so they can be used outside of The Loop as well?

Basically, we add a check at the beginning for a `post_id` and if there isn't one, we assume we are in The Loop and assign the variable with `get_the_ID()`. And then adjust the rest of the code to use the `post_id` variable instead. What do you think, @bacoords?
